.github/workflows/build-graphiql.yml

```diff
@@ -0,0 +1,40 @@
+# This workflow tracks changes to the "packages" and "build" directories
+# and if changes are detected, we run npm scripts to re-build the GraphiQL app
+# and commit the changes
+name: build-graphiql
+
+on:
+  push:
+    # Build GraphiQL only if changes were made to packages directory
+    paths:
+      - "packages/**"
+    branches:
+      - develop
+      - master
+
+jobs:
+  # Check to see if changes were made to the packages directory or the build directory in the PR
+  build-graphiql:
+    name: Build GraphiQL
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{ secrets.token }}
+
+      # Install and build the GraphiQL App
+      - name: Install NPM Dependencies and Build the GraphiQL App
+        id: install-and-build
+        run: |
+          npm ci && npm run build
+
+      # Commit the built assets for GraphiQL back to the repo
+      - name: Commit built GraphiQL
+        id: commit-changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Apply GraphiQL build changes
+          # Since the build directory is .gitignored, we force the changes
+          push_options: --force
```

Comment:
This feels wrong since it is under `push_options`.  I think you want to add the `--force` flag to `git add`.

Also not sure why the `stefanzweifel/git-auto-commit-action` is needed. You could just call the git cli directly:

```
git add --force builddir
git commit -m "Commit build"
git push origin HEAD:${{ github.head_ref }}
```